### ISSUE TITLE
AppCenter post build overrides custom URL Scheme defined in player settings in Unity

### ISSUE THREAD

**Issue by marcinwalus** [Microsoft/AppCenter-SDK-Unity#467]
### **Description**

For iOS I want to define custom URL scheme and I'm adding new entry to Player Settings->iOS->Other settings->Configuration section->Supported URL Schemes in Unity project. 
After iOS project is build only "appcenter-xxxxx" scheme is defined in Info.plist.

The reason is script AppCenterPostBuild.OnPostprocessInfo() which creates new array for "CFBundleURLTypes" instead of editing it if exists.

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. Create Unity project for iOS
2. Implement AppCenter Unity plugin and turn on Distribute feature

<img width="391" alt="ApPCenterSettings" src="https://user-images.githubusercontent.com/1926917/89683975-cd3ced80-d8f9-11ea-8273-faadf4a9e14c.png">

3. Define new URL scheme (ex. "myscheme") in Unity project settings: Player Settings->iOS->Other settings->Configuration section->Supported URL Schemes

<img width="590" alt="PlayerSettings" src="https://user-images.githubusercontent.com/1926917/89683981-d0d07480-d8f9-11ea-8f33-4bc21175d0f0.png">

4. Build iOS project

<img width="518" alt="InfoPlist" src="https://user-images.githubusercontent.com/1926917/89684418-af23bd00-d8fa-11ea-9556-76780d32420f.png">

Current result: Info.plist contains only one entry in "CFBundleURLTypes" array for "appcenter-xxxxx"
Expected result: Info.plist contains two "appcenter-xxxxx" and "myscheme" entries in "CFBundleURLTypes" array

### **Details**

1. Which SDK version are you using?
    - 3.3.0
2. Which platform did you experience the issue on?
    - iOS (with Distribute)
3. What device version did you see this error on?  Were you using an emulator or a physical device?
    - n/a
4. What third party plugins are you using?
    - n/a
5. Please enable verbose logging for your app using **Initial Log Level** setting and include the logs here:
    - n/a
6. Please share a screenshot of App Center Settings (without app secrets)
7. Please share a screenshot of Build and Player Settings.

### **Workaround**

Modify AppCenterPostBuild.OnPostprocessInfo() function and in "useDistribute" block add url schemes defined in player settings

```csharp
var schemes = PlayerSettings.iOS.iOSUrlSchemes;

foreach (var scheme in schemes)
{
    var urlType = urlTypes.GetType().GetMethod("AddDict").Invoke(urlTypes, null);
    var setStringMethod = urlType.GetType().GetMethod("SetString");
    setStringMethod.Invoke(urlType, new object[] { "CFBundleTypeRole", "None" });
    setStringMethod.Invoke(urlType, new object[] { "CFBundleURLName", ApplicationIdHelper.GetApplicationId() });
    var urlSchemes = urlType.GetType().GetMethod("CreateArray").Invoke(urlType, new[] { "CFBundleURLSchemes" });
    urlSchemes.GetType().GetMethod("AddString").Invoke(urlSchemes, new[] { scheme });
}
```

**Comment by vvechkanov** [comment]
Hi @marcinwalus!

Thank you for reporting this issue. I test it and it is reproducing correctly. I marked it as a bug, and add it to our team's board.

**Comment by AnastasiaKubova** [comment]
The fix has been released in App Center version [3.3.1](https://github.com/microsoft/appcenter-sdk-unity/releases/tag/3.3.1).
